server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteRestoreSnapshotIT.java

```diff
@@ -706,24 +751,169 @@ public void testInvalidRestoreRequestScenarios() throws Exception {
         indexDocuments(client, index, numDocsInIndex, numDocsInIndex + randomIntBetween(2, 5));
         ensureGreen(index);
 
-        // try index restore with remote store disabled
+        // try index restore with USER_UNREMOVABLE_SETTINGS setting disabled
         SnapshotRestoreException exception = expectThrows(
             SnapshotRestoreException.class,
             () -> client().admin()
                 .cluster()
                 .prepareRestoreSnapshot(snapshotRepo, snapshotName1)
                 .setWaitForCompletion(false)
-                .setIgnoreIndexSettings(SETTING_REMOTE_STORE_ENABLED)
+                .setIgnoreIndexSettings(IndexMetadata.SETTING_REMOTE_STORE_ENABLED)
                 .setIndices(index)
                 .setRenamePattern(index)
                 .setRenameReplacement(restoredIndex)
                 .get()
         );
         assertTrue(exception.getMessage().contains("cannot remove setting [index.remote_store.enabled] on restore"));
 
-        // try index restore with remote store repository modified
-        Settings remoteStoreIndexSettings = Settings.builder()
-            .put(IndexMetadata.SETTING_REMOTE_SEGMENT_STORE_REPOSITORY, newRemoteStoreRepo)
+        // try index restore with UnmodifiableOnRestore setting disabled
+        exception = expectThrows(
```

Comment:
We should add these tests in `RestoreSnapshotIT.java` instead.  These new tests are not specific to remote store. Also when you move lets create separate test methods for each scenarios.